src/network/networkd-lldp-tx.c

```diff
@@ -184,6 +195,37 @@ static int lldp_make_packet(
                 p = mempcpy(p, pretty_hostname, pretty_hostname_length);
         }
 
+        if (mud) {
+                _cleanup_free_ void *oui_mud = NULL;
+                /*
+                 * +--------+--------+----------+---------+--------------
+                 * |TLV Type|  len   |   OUI    |subtype  | MUDString
+                 * |  =127  |        |= 00 00 5E|  = 1    |
+                 * |(7 bits)|(9 bits)|(3 octets)|(1 octet)|(1-255 octets)
+                 * +--------+--------+----------+---------+--------------
+                 * where:
+
+                 * o  TLV Type = 127 indicates a vendor-specific TLV
+                 * o  len = indicates the TLV string length
+                 * o  OUI = 00 00 5E is the organizationally unique identifier of IANA
+                 * o  subtype = 1 (as assigned by IANA for the MUDstring)
+                 8 o  MUDstring = the length MUST NOT exceed 255 octets
+                */
+
+                r = lldp_write_tlv_header(&p, SD_LLDP_TYPE_PRIVATE, sizeof(SD_LLDP_OUI_MUD) + 1 + mud_length);
+                if (r < 0)
+                        return r;
+
+                oui_mud = memdup(SD_LLDP_OUI_MUD, sizeof(SD_LLDP_OUI_MUD));
+                if (!oui_mud)
+                        return log_oom();
+
+                p = mempcpy(p, (const void *) oui_mud, sizeof(SD_LLDP_OUI_MUD));
```

Comment:
the `(const void*)` cast appears unnecessary

Reply:
you canjust copy SD_LLDP_UI_MUD directly. Why copy it first via memdup? i.e. this should just work:

```c
p = mempcpy(p, SD_LLDP_OUI_MUD, sizeof(SD_LLDP_OUI_MUD));
```

Reply:
The Cent OS  CI was failing . It thinks the macro as 3 variables . So I had to come around this.
```
20:46:46 ../src/network/networkd-lldp-tx.c: In function ‘lldp_make_packet’:
20:46:46 ../src/network/networkd-lldp-tx.c:217:72: error: macro "__mempcpy" passed 5 arguments, but takes just 3
20:46:46                  p = mempcpy(p, SD_LLDP_OUI_MUD, sizeof(SD_LLDP_OUI_MUD));
20:46:46                                                                         ^
20:46:46 ../src/network/networkd-lldp-tx.c:217:19: error: assignment from incompatible pointer type [-Werror]
20:46:46                  p = mempcpy(p, SD_LLDP_OUI_MUD, sizeof(SD_LLDP_OUI_MUD));
20:46:46                    ^
20:46:46 cc1: all warnings being treated as errors
```

Reply:
hmm, looks like a compiler bug. but let's not do malloc here still, instead just allocate a local variable:

```c
uint8_t out_mud[sizeof(SD_LLDP_…)] = SD_LLDP_…;
```

doing malloc for 3 bytes sounds unnecessary